sdk/transport_policies/curl/src/az_curl.c

```diff
@@ -277,46 +291,47 @@ int32_t _az_curl_upload_read_callback(void * ptr, size_t size, size_t nmemb, voi
 /**
  * handles POST request. It handles seting up a body for request
  */
-AZ_NODISCARD az_result
-az_curl_send_upload_request(CURL * p_curl, az_http_request const * p_request) {
+static AZ_NODISCARD az_result
+_az_http_client_curl_send_upload_request(CURL * p_curl, az_http_request const * p_request) {
   AZ_CONTRACT_ARG_NOT_NULL(p_curl);
   AZ_CONTRACT_ARG_NOT_NULL(p_request);
 
   // Method
   az_span body = { 0 };
   int32_t required_size
       = az_span_length(p_request->_internal.body) + az_span_length(AZ_SPAN_ZEROBYTE);
-  AZ_RETURN_IF_FAILED(az_span_malloc(required_size, &body));
+  AZ_RETURN_IF_FAILED(_az_span_malloc(required_size, &body));
```

Comment:
Do we have to allocate and copy the body? Can we just pass the address of the request buffer directly to curl?